vault-api/apiclient.go

```diff
@@ -0,0 +1,29 @@
+package vaultapi
+
+type ApiClient struct {
+	vaultID       string
+	vaultURL      string
+	tokenProvider TokenProvider
+	logLevel      LogLevel
+	token         string
+}
+
+func (client *ApiClient) insert(records map[string]interface{}, options map[string]interface{}) (responseBody, error) {
+
+	return nil, nil
```

Comment:
if the implementation is planned, pls have a TODO comment in all the methods

Reply:
okay Saketh